File: api/Product-analysis-worker/event/ProductAnalysisEvent.py
from event.BaseEvent import BaseEvent


class ProductAnalysisEvent(BaseEvent):
    def __init__(self, product_id=None, name=None, description=None, categories=None, store=None, skus=None):
        self.product_id = product_id
        self.name = name
        self.description = description
        self.categories = categories if categories is not None else set()
        self.store = store
        self.skus = skus

    class ProductSku:
        def __init__(self, sku_code=None, name=None, price=None, variants=None, image=None):
            self.sku_code = sku_code
            self.name = name
            self.price = price
            self.image = image
            self.variants = variants if variants is not None else {}

    class Store:
        def __init__(self, id=None, name=None):
            self.id = id
            self.name = name

File: api/Product/src/main/java/com/lemoo/product/event/eventModel/ProductAnalysisEvent.java
/*
 *  ProductEvaluatedEvent
 *  @author: Minhhieuano
 *  @created 3/12/2025 3:26 AM
 * */


package com.lemoo.product.event.eventModel;

import lombok.*;

import java.util.Map;
import java.util.Set;

@EqualsAndHashCode(callSuper = true)
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class ProductAnalysisEvent extends Event {
    private String productId;
    private String name;
    private String description;
    private Set<String> categories;
    private Store store;
    private Set<ProductSku> skus;


    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    @Builder
    public static class ProductSku {
        private String skuCode;
        private String name;
        private Long price;
        private String image;
        private Map<String, String> variants;
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    @Builder
    public static class Store {
        private String id;
        private String name;
    }
}
